File: server-php/src/Model/Game.php
<?php

declare(strict_types=1);

namespace App\Model;

use Maxi\Attribute\MaxiField;
use Maxi\Attribute\MaxiType;

/**
 * Game DTO — mirrors the G:Game type in shared/sports.mxs.
 * Scores default to 0 in both PHP and the MAXI schema.
 */
#[MaxiType(alias: 'G', name: 'Game')]
class Game
{
    public function __construct(
        #[MaxiField(typeExpr: 'int')]
        public int $id = 0,

        #[MaxiField(typeExpr: 'T')]
        public ?Team $homeTeam = null,

        #[MaxiField(typeExpr: 'T')]
        public ?Team $awayTeam = null,

        #[MaxiField(annotation: 'date', required: true)]
        public string $date = '',

        #[MaxiField(typeExpr: 'enum[scheduled,live,finished,cancelled]')]
        public string $status = '',

        #[MaxiField(typeExpr: 'int', defaultValue: 0)]
        public int $homeScore = 0,

        #[MaxiField(typeExpr: 'int', defaultValue: 0)]
        public int $awayScore = 0,
    ) {}
}

File: server-java/src/main/java/io/maxi/examples/server/model/Game.java
package io.maxi.examples.server.model;

import io.maxi.annotation.MaxiField;
import io.maxi.annotation.MaxiType;

/**
 * Game DTO — mirrors the G:Game type in shared/sports.mxs.
 * Scores default to 0 in both Java and the MAXI schema.
 */
@MaxiType(alias = "G", name = "Game")
public class Game {

    @MaxiField(typeExpr = "int")
    public int id;

    @MaxiField(typeExpr = "T")
    public Team homeTeam;

    @MaxiField(typeExpr = "T")
    public Team awayTeam;

    @MaxiField(annotation = "date", required = true)
    public String date;

    @MaxiField(typeExpr = "enum[scheduled,live,finished,cancelled]")
    public String status;

    @MaxiField(typeExpr = "int", defaultValue = "0")
    public int homeScore;

    @MaxiField(typeExpr = "int", defaultValue = "0")
    public int awayScore;

    public Game() {}
}
